candy-machine-core/js/test/setup/txs-init.ts

```diff
@@ -0,0 +1,316 @@
+
+    const { nft: collection } = await metaplex
+      .nfts()
+      .create({
+        uri: COLLECTION_METADATA,
+        name: 'CORE Collection',
+        sellerFeeBasisPoints: 500,
+      })
+      .run();
+
+    const [, candyMachine] = await this.getKeypair('Candy Machine Account');
+    const authorityPda = findCandyMachineCreatorPda(candyMachine.publicKey, program.PROGRAM_ID);
+
+    await amman.addr.addLabel('Collection Mint', collection.address);
+
+    const collectionAuthorityRecord = findCollectionAuthorityRecordPda(
+      collection.mint.address,
+      authorityPda,
+    );
+    await amman.addr.addLabel('Collection Authority Record', collectionAuthorityRecord);
+
+    const collectionMetadata = findMetadataPda(collection.mint.address);
+    await amman.addr.addLabel('Collection Metadata', collectionMetadata);
+
+    const collectionMasterEdition = findMasterEditionV2Pda(collection.mint.address);
+    await amman.addr.addLabel('Collection Master Edition', collectionMasterEdition);
+
+    const accounts: program.InitializeInstructionAccounts = {
+      authorityPda,
+      collectionUpdateAuthority: collection.updateAuthorityAddress,
+      candyMachine: candyMachine.publicKey,
+      authority: payer.publicKey,
+      payer: payer.publicKey,
+      collectionMetadata,
+      collectionMint: collection.address,
+      collectionMasterEdition,
+      collectionAuthorityRecord,
+      tokenMetadataProgram: METAPLEX_PROGRAM_ID,
+      systemProgram: SystemProgram.programId,
+    };
+
+    const args: program.InitializeInstructionArgs = {
+      data: data,
+    };
+
+    const ixInitialize = program.createInitializeInstruction(accounts, args);
+    const ixCreateAccount = SystemProgram.createAccount({
+      fromPubkey: payer.publicKey,
+      newAccountPubkey: candyMachine.publicKey,
+      lamports: await connection.getMinimumBalanceForRentExemption(getCandyMachineSpace(data)),
+      space: getCandyMachineSpace(data),
+      programId: program.PROGRAM_ID,
+    });
+
+    const tx = new Transaction().add(ixCreateAccount).add(ixInitialize);
+
+    const txPromise = handler.sendAndConfirmTransaction(
+      tx,
+      [candyMachine, payer],
+      'tx: Initialize',
+    );
+
+    return { tx: txPromise, candyMachine: candyMachine.publicKey };
+  }
+
+  async addConfigLines(
+    t: Test,
+    candyMachine: PublicKey,
+    payer: Keypair,
+    lines: program.ConfigLine[],
+  ): Promise<{ txs: Transaction[] }> {
+    const accounts: program.AddConfigLinesInstructionAccounts = {
+      candyMachine: candyMachine,
+      authority: payer.publicKey,
+    };
+
+    const txs: Transaction[] = [];
+    let start = 0;
+
+    while (start < lines.length) {
+      // sends the config lines in chunks of 10
+      const limit = Math.min(lines.length - start, 10);
+      const args: program.AddConfigLinesInstructionArgs = {
+        configLines: lines.slice(start, start + limit),
+        index: start,
+      };
```

Comment:
Not neccesarily relevant here, but there are instances where extra long config lines will blow up the txn size limit even if you only send 10 lines at a time.

Reply:
We will add a note to the readme – this needs to be handled by the client.